### PR TITLE
Docker Documentation

### PR DESCRIPTION
**Pull request by r3d91ll**
Hey All,

I have updated the Docker instructions for the website. The following website documents have been updated:
Installation.md - changes here focus how to use the the Dockerfile.base and Dockerfile.full 
Contribute.md - This section focuses on how to use Dockerfile.dev to contribute to the project.

Additionally, changes were made to the Dockerfile.base, Dockerfile.full, and Dockerfile.dev files in autogen/samples/docker/ and I also created a Dockerfiles.md, which discusses how to use it for Development and how to adjust the Dockerfile to fit particular Dev environments.

## Why are these changes needed?

- Previous iterations of the Dockerfiles created environments that dropped the user into root. Developing as the root user should not be encouraged and can present security risks to applications built in such an environment. An autogen user is now created and given sudo access should root access be required.
- Additionally, I removed the virtual environment inside of docker as it is not needed.
- Added the installation of Vim nano and curl in the Dockerfile.dev for quality of life at the Docker CLI
- I added a commented block for the user to add environmental variables to their docker containers; this is useful for access keys and such.

## Related issue number
#1103 #1147 #1111
## Checks

- [x] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by r3d91ll** [comment]
@microsoft-github-policy-service agree

File: website/sidebars.js [hunk 1]
```diff
@@ -18,5 +18,9 @@
     'Research',
   ],
   // pydoc-markdown auto-generated markdowns from docstrings
-  referenceSideBar: [require("./docs/reference/sidebar.json")]
+  
+  // commenting out the reference sidebar allows the docs to be built 
+  // the reference sidebar is not found in the docs folder
+  // referenceSideBar: [require("./docs/reference/sidebar.json")]
+
```
Comment (sonichi) [inline website/sidebars.js:25]: Remove this change

File: website/docs/Installation.md [hunk 2]
```diff
@@ -1,226 +1,150 @@
...
-Example notebooks:
+### Optional Dependencies
 
-[Using MathChat to Solve Math Problems](https://github.com/microsoft/autogen/blob/main/notebook/agentchat_MathChat.ipynb)
+- **Docker**: Highly recommended for code execution. Install with `pip install docker`.
+- **Blendsearch**: For hyperparameter optimization. Install with `pip install "pyautogen[blendsearch]<0.2"`.
+- **Retrievechat**: For retrieval-augmented tasks. Install with `pip install "pyautogen[retrievechat]"`.
+- **Teachability**: For teachable agents. Install with `pip install "pyautogen[teachable]"`.
+- **LMM Agents**: For multimodal conversable agents. Install with `pip install "pyautogen[lmm]"`.
+- **Mathchat**: For math problem-solving agents. Install with `pip install "pyautogen[mathchat]<0.2"`.
```
Comment (sonichi) [inline website/docs/Installation.md:150]: Is it a good idea to remove the notebook examples?

**Comment by r3d91ll** [comment]
Hey all,

Ill close and re-submit with that edit removed. my apologies

**Comment by r3d91ll** [comment]
closing to address issues with websites/sidebars.js

**Comment by r3d91ll** [comment]
sorry about that, I have fixed the issues with the website/sidebards.js file.

File: website/sidebars.js (hunk 1 above)
Comment (r3d91ll) [reply website/sidebars.js]: Removed.

**Comment by qingyun-wu** [comment]
Could you fix the conflicts? Thank you!

File: website/docs/Installation.md [hunk 3]
```diff
@@ -1,226 +1,150 @@
...
-Example notebook:  [Chatting with a teachable agent](https://github.com/microsoft/autogen/blob/main/notebook/agentchat_teachability.ipynb)
+2. To exit the Poetry environment:
 
+   ```bash
+   exit
+   ```
 
+## Python Version Compatibility
 
-- #### Large Multimodal Model (LMM) Agents
```
Comment (qingyun-wu) [inline website/docs/Installation.md:205]: I believe we want to keep the advanced use case examples that require optional dependencies.

Comment (r3d91ll) [reply website/docs/Installation.md:205]: I agree, that was unintentional on my part. I will add that section and the others that were accidentally removed from the Installation.md file

Comment (r3d91ll) [reply website/docs/Installation.md]: Added the missing sections back. However, should we move the docker information under the Optional Dependencies to the Dockerfile.md?

File: website/docs/Installation.md (hunk 2 above)
Comment (r3d91ll) [reply website/docs/Installation.md]: removed and added the link back to the MathChat notebook

**Review by r3d91ll — COMMENTED** [review]
I think I cleaned up all the line and all accidental deletions have been put back in place.

File: website/docs/Contribute.md [hunk 4]
```diff
@@ -58,57 +57,84 @@ There is currently no formal reviewer solicitation process. Current reviewers id
...
+
+```bash
+docker run -it -p 8081:3000 -v `pwd`/autogen-newcode:newstuff/ autogen_dev_img bash
 ```
 
-Detailed instructions can be found [here](Installation.md#option-1-install-and-run-autogen-in-docker).
+- Note that the `pwd` is shorthand for present working directory. Thus, any path after the pwd is relative to that. If you want a more verbose method you could remove the "`pwd`/autogen-newcode" and replace it with the full path to your directory
 
+```bash
+docker run -it -p 8081:3000 -v `/home/AutoGenDeveloper/autogen-newcode:newstuff/ autogen_dev_img bash
```
Comment (qingyun-wu) [inline website/docs/Contribute.md:136]: Are we missing a ` or having an extra one here?

**Review by qingyun-wu — APPROVED** [review]
LGTM. Nice documentation! Thanks!

**Review by maxim-saplin — COMMENTED** [review]
Looks good!

**Pull request merged**